### ISSUE TITLE
Add option to nest components 

### ISSUE THREAD

**Issue by peterp** [redwoodjs/redwood#340]
We generate a lot of React components in `web/src/components`. Each scaffold command generates 5 components.

This could become problematic over time, and some people like to nest components by domain. We should come up with a structure and workflow that allows generated React components to be nested.

Based on feedback from #317 

> I prefer to have my components nested by domain, so when I generated yarn rw g scaffold post, I would love an option to have them generate in ./web/src/components/post/*/** to keep things organized better as my list of components grows.
> This would need to be supported for multiple fenerate commands (e.x. yarn rw g cell BlogPosts should have option for me to specify I want it to end up in a parent folder named components/blog/BlogPosts)

**Comment by skv-headless** [comment]
I would love to see something like this
<img width="226" alt="Screenshot 2020-04-13 at 20 22 39" src="https://user-images.githubusercontent.com/1488195/79137915-aa278480-7dc4-11ea-9f49-fdd8eea9eef2.png">

**Comment by jtoar** [comment]
WIP implementation at #423.

**Comment by jvanbaarsen** [comment]
Now that #423 is merged, I was wondering would it be an idea if you run `yarn rw g component Todo/Add` it would actually use the name of the component to see if there is a potential file structure in there (so the `/`). This would allow you to generate a loose component inside another component as parent.

So you would have something like:

```
- src
  - components
    - Todo
        - Todo.js
        - Todo.test.js
        - ...etc
        - Add
          - Add.js
          - Add.test.js
          - ...etc
```

**Comment by byrichardpowell** [comment]
**TLDR:** `yarn rw g component Todo/Add` would be very helpful

**Context:** it's common to have components that have their own children.  For example, imagine a `<ProductCategory/>`.  It renders a `<Heading/>` and some info about the category and then it maps it's products, rendering one `<ProductListItem/>` per product.  To achieve the approach I'm used to a folder structure like:

* /Components/Category
* /Components/Category/Category.tsx
* /Components/Category/Product/Product.tsx 

Right now this could be solved by doing something like 

* /Components/Category
* /Components/CategoryProduct

I _think_ this would get messy quite quickly though.

**Comment by jtoar** [comment]
Basic nesting is being added in #2430 and #2543.

**Comment by syabro** [comment]
Also I would like to see everything grouped by domain, since domain could contain pages, layouts and componets
